board/app/models/follow.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+class Follow extends AppModel 
+{
+
+    public function addFollowing()
+    { 
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+            'username' => $this->username,
+            'user_id' => $this->user_id
+            );
+            $db->insert('follow', $params);
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+
+    public function removeFollowing()
+    {
+        try {
+            $db = DB::conn();
+            $db->begin();
+            $params = array(
+                $this->username,
+                $this->user_id
+            );
+            $db->query('DELETE FROM follow WHERE username = ? AND user_id = ?', $params);
+            $db->commit();
+        } catch (Exception $e) {
+            $db->rollback();
+        }
+    }
+    
+    public function isUserFollowing()
+    {
+        $db = DB::conn();
+
+        if (!is_int($offset) || !is_int($limit)) { 
+            throw new NotIntegerException; 
+        }
+
+        $params = array(
+            $this->username,
+            $this->user_id
+        );
+        $user_following = $db->row('SELECT * FROM follow WHERE username = ? AND user_id = ?', $params);
+        return !$user_following;
+    }
+
+    public static function getDataByUserId($user_id)
+    {
+       return new self(objectToArray(User::getData($user_id)));
+    }
+
+    public static function getAllFollowing($offset, $limit, $user_id)
+    {
+        $following = array();
+        $db = DB::conn();
+
+        if (!is_int($offset) || !is_int($limit)) { 
+            throw new NotIntegerException; 
+        }
+                        
+        $rows = $db->rows("SELECT * FROM follow WHERE user_id = ? LIMIT {$offset}, {$limit}", array($user_id));
+
+        foreach($rows as $row) {
+            $following[] = new self($row);
+        }
+        return $following;
+    }
+
+    public static function getRecentActivity($offset, $limit, $user_id)
+    {
+        $home = array();
+        $db = DB::conn();
+                        
+        $rows = $db->rows("SELECT * FROM follow WHERE user_id = ?", array($user_id));
+
+        foreach($rows as $row) {
+            $home[] = new self($row);
+        }
+        return $home;
+    }
+
+    public static function countFollowing($user_id)
+    {
+        $db = DB::conn();
+        $total_following = $db->value('SELECT COUNT(*) FROM follow WHERE user_id = ?', array($user_id));
+        
+        return $total_following;
+    }
+
+    public function countOtherUsers($user_id)
+    {
+        return User::countOtherUser($this->user_id);
+    }
+
+    public static function countNewsfeed($user_id) 
+    {
+        $db = DB::conn();
+        $users = $db->value('SELECT COUNT(*) FROM follow WHERE user_id = ?', array($user_id));
+        
+        return $users;
```

Comment:
The return type "$users" is a bit confusing since the function seems to 'count the news feed'? So either rename the function if you are returning an instance of a 'user', or rename this variable :bow: